sklearn/tree/tree.py

```diff
@@ -240,8 +256,17 @@ def fit(self, X, y, sample_weight=None, check_input=True,
                              "number of samples=%d" % (len(y), n_samples))
         if self.min_samples_split <= 0:
             raise ValueError("min_samples_split must be greater than zero.")
+        if not 1 < min_samples_split <= n_samples:
+            raise ValueError("min_samples_split must in [2, n_samples]")
         if self.min_samples_leaf <= 0:
             raise ValueError("min_samples_leaf must be greater than zero.")
+        elif 0.5 <= self.min_samples_leaf < 1:
```

Comment:
@MechCoder It is the fraction of samples that should at least exist in a leaf when you make a split. Since when you split, you are making two leaves, it is impossible to put more than 50% of the samples in both of them (since otherwise the sum would be larger than 100%). 

However, `min_samples_leaf=0.5` could be a valid value.